src/app/pages/datasets/components/dataset-form/dataset-form.component.ts

```diff
@@ -175,13 +175,11 @@ export class DatasetFormComponent implements OnInit {
       : this.ws.call('pool.dataset.update', [this.existingDataset.id, payload as DatasetUpdate]);
 
     request$.pipe(
-      switchMap(
-        (dataset) => this.checkForAclOnParent(dataset),
-        (dataset) => dataset,
-      ),
+      switchMap((dataset) => forkJoin([of(dataset), this.checkForAclOnParent()])),
+      switchMap(([dataset, isAcl]) => forkJoin([of(dataset), isAcl ? this.aclDialog() : of(false)])),
```

Comment:
It's a bit hard to follow. Let's do something else, maybe let's switch to an inner observable and then use `switchMap` and `map` on the inner observable.